### PR TITLE
Feature/add nft layouts

### PR DESCRIPTION
**Pull request by RustemYuzlibaev**
[PW-512)]( https://soramitsu.atlassian.net/browse/PW-512?atlOrigin=eyJpIjoiNTI5YjU1YWE2MzhiNDgzNGI5MjllOTZhNmU1NDhiNzQiLCJwIjoiaiJ9 )

### CODE REVIEW

File: src/components/CreateNFT.vue [hunk 1]
```diff
@@ -0,0 +1,454 @@
...
+  }
+
+  &__placeholder {
+    letter-spacing: var(--s-letter-spacing-small);
+    color: var(--s-color-base-content-secondary);
+    font-weight: 400 !important;
+    font-size: calc(var(--s-size-small) / 2);
+    text-align: center;
+    padding: 0 50px;
+    color: #2a171f;
```
Comment (Nikita-Polyakov) [inline src/components/CreateNFT.vue:439]: `var(--s-color-base-content-primary)` is used in figma and looks good using dark theme too

nft-dropbox

File: src/util/index.ts [hunk 2]
```diff
@@ -128,6 +130,42 @@ export const toHashTable = (list: Array<any>, key: string) => {
   }, {});
 };
 
+export const fileToBase64 = (file: File): Promise<string | null> =>
```
Comment (stefashkaa) [inline src/util/index.ts:133]: may you create a class with static methods like 
```
export class IpfsStorage {
  static fileToBase64() ...
}
```

Comment (stefashkaa) [reply src/util/index.ts:133]: it can be a separated file

File: src/components/CreateNFT.vue [hunk 3]
```diff
@@ -0,0 +1,454 @@
...
+    this.imageLoading = true;
+    this.badSource = false;
+    this.linkPlaceholder = link
+      ? this.t('createToken.nft.link.placeholderShort')
+      : this.t('createToken.nft.link.placeholder');
+
+    this.checkImageFromSource(link);
+  }
+
+  async checkImageFromSource(url: string): Promise<void> {
```
Comment (Nikita-Polyakov) [inline src/components/CreateNFT.vue:240]: I've tried to upload a random image by link, but access is blocked by CORS. Looks like we should handle this error, and stop showing loader inside dropbox area

nft-upload-link

File: src/components/CreateNFT.vue [hunk 4]
```diff
@@ -0,0 +1,454 @@
...
+  .placeholder {
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+  }
+
+  &__content {
+    height: 176px;
+    width: 176px;
+    object-fit: cover;
```
Comment (Nikita-Polyakov) [inline src/components/CreateNFT.vue:422]: With fixed width & height wide image looks cropped. Maybe better to use `object-fit: contain` ?

File: src/components/CreateNFT.vue (hunk 1 above)
Comment (RustemYuzlibaev) [reply src/components/CreateNFT.vue:439]: fixed

File: src/util/index.ts (hunk 2 above)
Comment (RustemYuzlibaev) [reply src/util/index.ts:133]: Removed to separate file

File: src/components/CreateNFT.vue (hunk 3 above)
Comment (RustemYuzlibaev) [reply src/components/CreateNFT.vue:240]: Fixed. Wrapped in try/catch block

File: src/components/CreateNFT.vue (hunk 4 above)
Comment (RustemYuzlibaev) [reply src/components/CreateNFT.vue:422]: Tried that too. The reason behind it if user provides image not appropriate in size it looks very small. 
In other NFT platforms, by the way, they usually tell recommended image size. Don't know what to use better here

https://user-images.githubusercontent.com/27702061/149295819-505c3fd9-a8da-45a4-8c4e-5f6153b13e5b.mov

File: src/store/Settings.ts [hunk 5]
```diff
@@ -29,23 +30,32 @@ const types = flow(
...
     'SET_NETWORK_FEES',
     'RESET_RUNTIME_VERSION_SUBSCRIPTION',
+    'SET_NFT_STORAGE',
+    'SET_API_KEYS',
   ]),
   map((x) => [x, x]),
   fromPairs
 )([]);
 
+export type ApiKeysObject = {
```
Comment (Nikita-Polyakov) [inline src/store/Settings.ts:40]: Maybe we should add this type to `WALLET_TYPES ` to reuse it polkaswap?

**Review by Nikita-Polyakov — APPROVED** [review]

File: src/store/Settings.ts (hunk 5 above)
Comment (RustemYuzlibaev) [reply src/store/Settings.ts]: Moved to WALLET_TYPES

File: src/api/index.ts [hunk 6]
```diff
@@ -1,5 +1,4 @@
 import { api, connection } from '@sora-substrate/util';
-
```
Comment (stefashkaa) [inline src/api/index.ts:2]: please discard that change

**Pull request merged**